<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>viewer</class>
 <widget class="QMainWindow" name="viewer">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>856</width>
    <height>720</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="3" column="1" colspan="2">
     <layout class="QHBoxLayout" name="projection_type">
      <item>
       <widget class="QRadioButton" name="parallel">
        <property name="text">
         <string>parallel</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">model_type</string>
        </attribute>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="central">
        <property name="text">
         <string>central</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">model_type</string>
        </attribute>
       </widget>
      </item>
     </layout>
    </item>
    <item row="5" column="1" colspan="2">
     <layout class="QHBoxLayout" name="background_color">
      <item>
       <widget class="QLabel" name="back_color">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="choose_back_color">
        <property name="text">
         <string>Choose</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">chooser</string>
        </attribute>
       </widget>
      </item>
     </layout>
    </item>
    <item row="4" column="1" colspan="2">
     <widget class="QLabel" name="label">
      <property name="font">
       <font>
        <pointsize>13</pointsize>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Background color</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QLabel" name="edges_number">
      <property name="text">
       <string>0</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QLabel" name="opened_file">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="maximumSize">
       <size>
        <width>200</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="text">
       <string>-</string>
      </property>
      <property name="wordWrap">
       <bool>true</bool>
      </property>
     </widget>
    </item>
    <item row="9" column="1" colspan="2">
     <layout class="QHBoxLayout" name="thickness_layout">
      <item>
       <widget class="QLabel" name="thikness">
        <property name="text">
         <string>Thickness</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="thickness_size">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="value">
         <number>5</number>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="1" column="2">
     <widget class="QLabel" name="edges">
      <property name="text">
       <string>edges</string>
      </property>
     </widget>
    </item>
    <item row="6" column="1" colspan="2">
     <widget class="QLabel" name="edges_title">
      <property name="font">
       <font>
        <pointsize>13</pointsize>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Edges</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item row="7" column="1" colspan="2">
     <layout class="QHBoxLayout" name="edges_layout">
      <item>
       <widget class="QLabel" name="edge_color">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="choose_edges_color">
        <property name="text">
         <string>Choose</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">chooser</string>
        </attribute>
       </widget>
      </item>
     </layout>
    </item>
    <item row="13" column="1" colspan="2">
     <layout class="QHBoxLayout" name="size_layout">
      <item>
       <widget class="QLabel" name="size">
        <property name="text">
         <string>Size</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="size_num">
        <property name="value">
         <number>10</number>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="14" column="1" colspan="2">
     <layout class="QHBoxLayout" name="rotate_layout">
      <item>
       <widget class="QDoubleSpinBox" name="rotate_spin">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimum">
         <double>-360.000000000000000</double>
        </property>
        <property name="maximum">
         <double>360.000000000000000</double>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="rotate_combo">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <item>
         <property name="text">
          <string>X</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Y</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Z</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="rotate">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Rotate</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="11" column="1" colspan="2">
     <layout class="QHBoxLayout" name="vertices_type">
      <item>
       <widget class="QRadioButton" name="none">
        <property name="text">
         <string>none</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
        <property name="autoExclusive">
         <bool>false</bool>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">vertices_btn</string>
        </attribute>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="circle">
        <property name="text">
         <string>circle</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">vertices_btn</string>
        </attribute>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="sqare">
        <property name="text">
         <string>square</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">vertices_btn</string>
        </attribute>
       </widget>
      </item>
     </layout>
    </item>
    <item row="8" column="1" colspan="2">
     <layout class="QHBoxLayout" name="solid_dotted">
      <item>
       <widget class="QRadioButton" name="solid">
        <property name="text">
         <string>Solid</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">edges_type</string>
        </attribute>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="dotted">
        <property name="text">
         <string>Dotted</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">edges_type</string>
        </attribute>
       </widget>
      </item>
     </layout>
    </item>
    <item row="10" column="1" colspan="2">
     <widget class="QLabel" name="vertices_title">
      <property name="font">
       <font>
        <pointsize>13</pointsize>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Vertices</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
    <item row="0" column="2">
     <widget class="QPushButton" name="open_file">
      <property name="text">
       <string>Open</string>
      </property>
     </widget>
    </item>
    <item row="17" column="1" colspan="2">
     <widget class="QPushButton" name="save">
      <property name="text">
       <string>Save</string>
      </property>
     </widget>
    </item>
    <item row="12" column="1" colspan="2">
     <layout class="QHBoxLayout" name="vertices_layout">
      <item>
       <widget class="QLabel" name="vertices_color">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="choose_vertices_color">
        <property name="text">
         <string>Choose</string>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">chooser</string>
        </attribute>
       </widget>
      </item>
     </layout>
    </item>
    <item row="2" column="2">
     <widget class="QLabel" name="vertexes">
      <property name="text">
       <string>vertexes</string>
      </property>
     </widget>
    </item>
    <item row="15" column="1" colspan="2">
     <layout class="QHBoxLayout" name="move_layout">
      <item>
       <widget class="QDoubleSpinBox" name="move_spin">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimum">
         <double>-100.000000000000000</double>
        </property>
        <property name="maximum">
         <double>100.989999999999995</double>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="move_combo">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <item>
         <property name="text">
          <string>X</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Y</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Z</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="move">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Move</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="16" column="1" colspan="2">
     <layout class="QHBoxLayout" name="scale_layout">
      <item>
       <widget class="QDoubleSpinBox" name="scale_spin">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimum">
         <double>0.100000000000000</double>
        </property>
        <property name="maximum">
         <double>100.000000000000000</double>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="scale">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Scale</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="2" column="1">
     <widget class="QLabel" name="vertices_number">
      <property name="text">
       <string>0</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
    </item>
    <item row="0" column="0" rowspan="19">
     <widget class="s21::OpenGLWidget" name="open_gl">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
    <item row="18" column="1" colspan="2">
     <widget class="QPushButton" name="screencast">
      <property name="text">
       <string>Screencast</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>s21::OpenGLWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>OpenGLWidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="chooser"/>
  <buttongroup name="vertices_btn"/>
  <buttongroup name="model_type"/>
  <buttongroup name="edges_type"/>
 </buttongroups>
</ui>
